src/legacy/ui/public/agg_types/buckets/date_histogram.js

```diff
@@ -155,16 +155,25 @@ export const dateHistogramBucketAgg = new BucketAggType({
     },
     {
       name: 'time_zone',
-      default: () => {
-        const isDefaultTimezone = config.isDefault('dateFormat:tz');
-        return isDefaultTimezone ? detectedTimezone || tzOffset : config.get('dateFormat:tz');
-      },
-      serialize() {
-        // We don't want to store the `time_zone` parameter ever in the saved object for the visualization.
-        // If we would store this changing the time zone in Kibana would not affect any already saved visualizations
-        // anymore, which is not the desired behavior. So always returning undefined here, makes sure we're never
-        // saving that parameter and just keep it "transient".
-        return undefined;
+      default: undefined,
+      // We don't ever want this parameter to be serialized out (when saving or to URLs)
+      // since we do all the logic handling it "on the fly" in the `write` method, to prevent
+      // time_zones being persisted into saved_objects
+      serialize: () => undefined,
+      write: (agg, output) => {
+        // If a time_zone has been set explicitly always prefer this.
+        let tz = agg.params.time_zone;
```

Comment:
Is this code path currently in use? Or is this just here to support editors which might set the time_zone param?

Reply:
It's currently not in use anywhere, but since you could configure a time_zone via manually editing expression (or using the expressions via an API) I wanted to leave that in (especially since we already had a test that checks for that behavior).

Reply:
OK, nice.